Minor fixes in packaging to make it compatible with maturin 1.4.0

johandahlberg opened the pull request:
I had some issues when trying to build `graspologic-native`, and the following updates seemed to do the trick to get everything working.

I basically:
 - reorganized the metadata a bit
 - relaxed the maturin requierment
 - update the install instructions
 - updated supported python versions in the README to match what is mentioned in the "classifier" section

Thank you for providing such a nice package!

johandahlberg commented:
@microsoft-github-policy-service agree company="Pixelgen Technologies AB"

daxpryce commented:
Single comment for review left inline, if 3.12 does break it then let's not worry about it now we'll just create an issue to fix it for later :)

daxpryce commented:
Seems like it is failing right now because maturin's cli args changed since the time I wrote it (which is why I had it pinned to the [0.12..0.13) range, I was tired of breaking changes. Once that is fixed in the github action for building it, we should be good to go though.

daxpryce commented:
https://github.com/johandahlberg/graspologic-native/blob/packaging-fixes/.github/workflows/build.yml#L76

daxpryce commented:
You will also need to bump the version in pyo3's Cargo.toml file to at least 1.2.1 (we can't publish over old versions and the build is set to fail if any of the publication steps fails). I don't think it's really worth trying to make Cargo happy with the pep rules around "1.2.0.post1" as a valid version, and we're not changing anything in the api so 1.2.1 should comply enough with semver to make everyone as happy as they're going to get :)

johandahlberg commented:
Thank you for the review!

I can't seem to find the inline comment you mention.

For the other issues I have:
 - changed the build actions to use `PyO3/maturin-action@v1` which as far as I've been able to tell is the new version of that used to be `messense/maturin-action@v1`. And then used that to set the target for the MacOS build.
 - explicitly adding build the source dist for the ubuntu build, since otherwise it didn't appear to show up
 - updated the version to ` 1.2.1` as suggested
 - locked the maturin version to be `[1.4, 2.0)`, with `1.4.0` specifically being used in the github workflow
 - updated a bunch of "base" actions to their latest versions to quite the warnings associated with them

Let me know if there is anything I've missed, or that you'd like done differently. Also, feel free to push any changes you deem necessary to this branch.

daxpryce commented:
My inline comments have fully disappeared. Let me do this again and check. In the meantime I'm going to approve the workflow and make sure it's working e2e.

daxpryce commented on .github/workflows/build.yml:
```diff
@@ -102,20 +104,20 @@ jobs:
           path: dist/
       - name: Generate SHA256 files for each wheel
         run: |
           sha256sum dist/*.whl > checksums.txt
```
can you also have it do `sha256sum dist/*.tar.gz >> checksums.txt`? It lets us match that what we built on our github action runners actually matches what is on pypi.org on publication :)

daxpryce commented on packages/pyo3/pyproject.toml:
```diff
@@ -1,3 +1,25 @@
+name = "graspologic-native"
+maintainer = "Dax Pryce"
+maintainer-email = "daxpryce@microsoft.com"
+requires-python = ">=3.6,<3.12"
```
If you want to, we can try to extend this to 3.12 too. It's currently in final so we can probably extend this and see how it works (the build matrix in build.yaml will need updated too to include 3.12). If it fails just back it out, but we should try it while we're in here :)

daxpryce reviewed (approved):
Approved!

daxpryce commented:
1.2.1 is now published to PyPI.  https://pypi.org/project/graspologic-native/1.2.1/#files

On Jan 23, 2024, at 11:00 PM, Johan Dahlberg ***@***.***> wrote:



Thank you for finishing this up! And, no worries about the inline comment - I didn't spend long looking for it.

—
Reply to this email directly, view it on GitHub<https://github.com/microsoft/graspologic-native/pull/39#issuecomment-1907483698>, or unsubscribe<https://github.com/notifications/unsubscribe-auth/AAG3SUZJY5LXKDCKLDS3PWLYQCWP5AVCNFSM6AAAAABB47YRX6VHI2DSMVQWIX3LMV43OSLTON2WKQ3PNVWWK3TUHMYTSMBXGQ4DGNRZHA>.
You are receiving this because you modified the open/close state.Message ID: ***@***.***>

johandahlberg commented:
Thank you!
